ext/node/ops/http2.rs

```diff
@@ -466,7 +466,9 @@ fn poll_data_or_trailers(
     }
     if let Poll::Ready(data) = body.poll_data(cx) {
       if let Some(data) = data {
-        return Poll::Ready(Ok(DataOrTrailers::Data(data?)));
+        let data = data?;
+        body.flow_control().release_capacity(data.len())?;
+        return Poll::Ready(Ok(DataOrTrailers::Data(data)));
       }
       // If data is None, loop one more time to check for trailers
       continue;
```

Comment:
I have a question (unrelated to the PR) - why can't we return `Poll::Pending` here and remove the outer loop? I see that this method is only used with `poll_fn`. This way we can keep this method loop free.

Reply:
yes, we can remove that.

Reply:
1a33977